### PR TITLE
mysqlctl: attempt to fix MySQL replication bug

### LINKED ISSUE #5067: repairReplication: attempt to fix 'Slave failed to initialize relay log info structure from the repository (errno 1872)'
There's an acknowledged [upstream MySQL bug](https://bugs.mysql.com/bug.php?id=83713) that affects at least 5.7 and 8.0 that shows this error when trying to start replication. We encounter it a few times per month.
`Slave failed to initialize relay log info structure from the repository (errno 1872)`

This has also been a cause for flaky tests in the past, so we may be experiencing it in CI more often than we know. #3237 

There is also a simple solution that should be 100% safe and works almost every time when we apply it manually.

    reset slave;
    start slave IO_THREAD;
    stop slave IO_THREAD;
    reset slave;
    start slave;

I'd like for `repairReplication` to attempt that when it encounters this specific error. If nobody is opposed, I'm happy to send in a PR.

@sjmudd has also experienced this issue and has commented on the linked MySQL bugs.

Here are the full Vitess logs surrounding the error. 

```
replication_reporter.go:69] Slave is stopped. Trying to reconnect to master...
locks.go:359] Locking shard domains/c0-e0 for action repairReplication to uscentral1-0498823900 as parent)
query.go:73] exec CHANGE MASTER TO ...
query.go:73] exec START SLAVE
locks.go:396] Unlocking shard domains/c0-e0 for action repairReplication to uscentral1-0498823900 as parent) with error ExecuteFetch(START SLAVE) failed: Slave failed to initialize relay log info structure from the repository (errno 1872) (sqlstate HY000) during query: START SLAVE
replication_reporter.go:72] Failed to reconnect to master: ExecuteFetch(START SLAVE) failed: Slave failed to initialize relay log info structure from the repository (errno 1872) (sqlstate HY000) during query: START SLAVE
```

### PR DESCRIPTION
**Pull request by derekperkins**
Fixes #5067

### CODE REVIEW

**Comment by derekperkins** [comment]
cc @sjmudd

File: go/vt/mysqlctl/query.go [hunk 1]
```diff
@@ -71,13 +71,54 @@ func (mysqld *Mysqld) ExecuteSuperQueryList(ctx context.Context, queryList []str
 func (mysqld *Mysqld) executeSuperQueryListConn(ctx context.Context, conn *dbconnpool.PooledDBConnection, queryList []string) error {
 	for _, query := range queryList {
 		log.Infof("exec %v", redactMasterPassword(query))
-		if _, err := mysqld.executeFetchContext(ctx, conn, query, 10000, false); err != nil {
+		_, err := mysqld.executeFetchContext(ctx, conn, query, 10000, false)
+		switch {
+		case err == nil:
+			return nil
+
+		case strings.ToUpper(query) == "START SLAVE" && isReplicationErr1872(err):
```
Comment (enisoc) [inline go/vt/mysqlctl/query.go]: I don't think a generic "execute query" function is the right place for this specific logic. If there's not a subroutine like `mysqld.StartSlave()` that all the code paths you're trying to catch use, then we should add one and change the appropriate sites to use it.

Or alternatively, would it work for you if we only check for this error in the repair section of the replication health reporter? That seems safer than injecting the fix at this low level, unless for some reason the periodic repair doesn't work for your needs.

Comment (derekperkins) [reply go/vt/mysqlctl/query.go]: Yeah, it felt weird to do this at this low level. I started at `repairReplication` and kept digging. That code path never called `StartSlave` that I found.

Comment (enisoc) [reply go/vt/mysqlctl/query.go]: If you're ok with only doing this check in the `repairReplication` code path, then we can just make sure sufficient error detail gets propagated back up to that level to detect this specific case, and put the logic there. WDYT?

Comment (derekperkins) [reply go/vt/mysqlctl/query.go]: That works for me. This was just a quick stab in the dark to get some feedback.

**Comment by deepthi** [comment]
I have tried to fix this at a different place in #5627

**Comment by yzsfly** [comment]
Cannot find the fix code in the branch 12.0.0

**Pull request closed without merging**